hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestSnapshotDeletingService.java

```diff
@@ -0,0 +1,465 @@
+        "snap1");
+
+    // Delete 5 Keys
+    for (int i = 1; i <= 5; i++) {
+      client.getProxy().deleteKey(VOLUME_NAME, BUCKET_NAME_TWO,
+          "key" + i, false);
+    }
+    // Rename Keys 3 keys
+    for (int i = 6; i <= 8; i++) {
+      client.getProxy().renameKey(VOLUME_NAME, BUCKET_NAME_TWO, "key" + i,
+          "renamedKey" + i);
+    }
+    // Delete 2 Dirs
+    for (int i = 1; i <= 2; i++) {
+      client.getProxy().deleteKey(VOLUME_NAME, BUCKET_NAME_TWO, "/parent" + i,
+          true);
+    }
+
+    assertTableRowCount(renamedKeyTable, 3);
+    // Delete Renamed Keys
+    for (int i = 6; i <= 8; i++) {
+      client.getProxy().deleteKey(VOLUME_NAME, BUCKET_NAME_TWO,
+          "renamedKey" + i, false);
+    }
+
+    assertTableRowCount(deletedTable, 8);
+    assertTableRowCount(deletedDirTable, 2);
+    assertTableRowCount(dirTable, 10);
+    assertTableRowCount(renamedKeyTable, 3);
+
+    // Create Snapshot2
+    client.getObjectStore().createSnapshot(VOLUME_NAME, BUCKET_NAME_TWO,
+        "snap2");
+
+    // Once snapshot is taken renamedTable, deletedTable should be cleaned
+    assertTableRowCount(renamedKeyTable, 0);
+    assertTableRowCount(deletedTable, 0);
+    // TODO: [SNAPSHOT] Once HDDS-8064 is done, This should be
+    // zero. As there shouldn't be directory expansion happening.
+    assertTableRowCount(deletedDirTable, 2);
+
+    // Delete 2 more keys
+    for (int i = 9; i <= 10; i++) {
+      client.getProxy().deleteKey(VOLUME_NAME, BUCKET_NAME_TWO,
+          "key" + i, false);
+    }
+
+    assertTableRowCount(deletedTable, 2);
+
+    // Create Snapshot3
+    client.getObjectStore().createSnapshot(VOLUME_NAME, BUCKET_NAME_TWO,
+        "snap3");
+
+    assertTableRowCount(renamedKeyTable, 0);
+    assertTableRowCount(deletedDirTable, 2);
+    assertTableRowCount(deletedTable, 0);
+    assertTableRowCount(keyTable, 9);
+    SnapshotInfo deletedSnap = om.getMetadataManager()
+        .getSnapshotInfoTable().get("/vol1/bucket2/snap2");
+
+    client.getObjectStore().deleteSnapshot(VOLUME_NAME, BUCKET_NAME_TWO,
+        "snap2");
+
+    // Once all the tables are moved, the snapshot is deleted
+    assertTableRowCount(om.getMetadataManager().getSnapshotInfoTable(), 2);
+
+    verifySnapshotChain(deletedSnap, "/vol1/bucket2/snap3");
+    OmSnapshot snap3 = (OmSnapshot) om.getOmSnapshotManager()
+        .checkForSnapshot(VOLUME_NAME, BUCKET_NAME_TWO,
+            getSnapshotPrefix("snap3"));
+
+    Table<String, OmKeyInfo> snapDeletedDirTable =
+        snap3.getMetadataManager().getDeletedDirTable();
+    Table<String, String> snapRenamedTable =
+        snap3.getMetadataManager().getSnapshotRenamedKeyTable();
+    Table<String, RepeatedOmKeyInfo> snapDeletedTable =
+        snap3.getMetadataManager().getDeletedTable();
+
+    assertTableRowCount(snapRenamedTable, 3);
+    assertTableRowCount(snapDeletedDirTable, 2);
+    // All the keys deleted before snapshot2 is moved to snap3
```

Comment:
I don't see any tests that confirm a successful move back to the active fs.  Am I missing those?

Reply:
Updated it.